### PR TITLE
params, cmd/utils, mobile: remove DiscoveryV5Bootnodes

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
As our regular bootnodes are started with protocol v5 enabled, we don't need to make the distinction between v5 and normal bootnodes anymore. 

Regarding mobile, I will add a new mechanism to get the bootnodes for the testnets in the future.

### CODE REVIEW

**Comment by MariusVanDerWijden** [comment]
Also all servers previously listed in DiscoveryV5Bootnodes seem to be down anyway

**Comment by karalabe** [comment]
Next question re the PR, although we remove the old bootnodes, we still retained the flags. Should we perhaps also get rid of those + the config separations for the V4 and V5? I think it would make sense, but we might break someone's config files since those are auto generated. CC @fjl

**Comment by adamschmideg** [comment]
Deprecate both flags and remove them later. The config options should remain.

**Comment by MariusVanDerWijden** [comment]
Put on hold until: https://github.com/ethereum/go-ethereum/pull/20935

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -793,9 +783,11 @@ func setNodeUserIdent(ctx *cli.Context, cfg *node.Config) {
...
 	urls := params.MainnetBootnodes
 	switch {
-	case ctx.GlobalIsSet(BootnodesFlag.Name) || ctx.GlobalIsSet(BootnodesV4Flag.Name):
-		if ctx.GlobalIsSet(BootnodesV4Flag.Name) {
-			urls = splitAndTrim(ctx.GlobalString(BootnodesV4Flag.Name))
+	case ctx.GlobalIsSet(BootnodesFlag.Name) || ctx.GlobalIsSet(LegacyBootnodesV4Flag.Name) || ctx.GlobalIsSet(LegacyBootnodesV5Flag.Name):
+		if ctx.GlobalIsSet(LegacyBootnodesV4Flag.Name) {
+			urls = splitAndTrim(ctx.GlobalString(LegacyBootnodesV4Flag.Name))
+		} else if ctx.GlobalIsSet(LegacyBootnodesV5Flag.Name) {
+			urls = splitAndTrim(ctx.GlobalString(LegacyBootnodesV5Flag.Name))
```
Comment (karalabe) [inline cmd/utils/flags.go:790]: I'm unsure about this. Originally we have v4, that could override the non versioned for 4, and v5 that could override the non versioned for 5. The new behavior is a bit dangerous, as specifying v5 could also override v4.

What was the reason for this change if we are keeping the flags as deprecated anyway?

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -805,7 +797,7 @@ func setBootstrapNodes(ctx *cli.Context, cfg *p2p.Config) {
 		urls = params.RinkebyBootnodes
 	case ctx.GlobalBool(GoerliFlag.Name):
 		urls = params.GoerliBootnodes
-	case cfg.BootstrapNodes != nil:
+	case cfg.BootstrapNodes != nil || cfg.BootstrapNodesV5 != nil:
```
Comment (karalabe) [inline cmd/utils/flags.go:800]: This seems dangerous. If either of the fields have been already set, you are skipping setting the other too. The original code had 2 methods, so you could skip setting v4 or v5 individually.

File: cmd/utils/flags.go [hunk 3]
```diff
@@ -820,36 +810,18 @@ func setBootstrapNodes(ctx *cli.Context, cfg *p2p.Config) {
...
-	}
 
-	cfg.BootstrapNodesV5 = make([]*discv5.Node, 0, len(urls))
-	for _, url := range urls {
-		if url != "" {
-			node, err := discv5.ParseNode(url)
-			if err != nil {
-				log.Error("Bootstrap URL invalid", "enode", url, "err", err)
-				continue
+	if ctx.GlobalIsSet(LegacyBootnodesV5Flag.Name) {
```
Comment (karalabe) [inline cmd/utils/flags.go:814]: This path will still not enter if the check exists on line 799

Comment (karalabe) [reply cmd/utils/flags.go:814]: If you want to merge the 2 methods, you'll need to handle the reassignment differently:

```go
if cfg.BootstrapNodes == nil {
    // Do code above
}
if cfg.BootstrapNodesV5 != nil {
    // Do code below
}
```

Comment (MariusVanDerWijden) [reply cmd/utils/flags.go:814]: Ahhh now I get it, thanks!

Comment (MariusVanDerWijden) [reply cmd/utils/flags.go]: fixed in https://github.com/ethereum/go-ethereum/pull/20949/commits/9e6dca0d599f6c6b97e04835ae4157a51607d2e0

Comment (MariusVanDerWijden) [reply cmd/utils/flags.go]: wait... why == in the first command and != in the second?

Comment (karalabe) [reply cmd/utils/flags.go]: copy paste error :) should have been == in the second too

File: cmd/utils/flags.go [hunk 4]
```diff
@@ -805,51 +795,34 @@ func setBootstrapNodes(ctx *cli.Context, cfg *p2p.Config) {
...
-		if ctx.GlobalIsSet(BootnodesV5Flag.Name) {
-			urls = splitAndTrim(ctx.GlobalString(BootnodesV5Flag.Name))
-		} else {
-			urls = splitAndTrim(ctx.GlobalString(BootnodesFlag.Name))
 		}
-	case ctx.GlobalBool(RinkebyFlag.Name):
-		urls = params.RinkebyBootnodes
-	case ctx.GlobalBool(GoerliFlag.Name):
-		urls = params.GoerliBootnodes
-	case cfg.BootstrapNodesV5 != nil:
```
Comment (karalabe) [inline cmd/utils/flags.go:840]: What happened to all these deleted cases? Previously we used the default bootnodes for mainnet and rinkeby and goerli (ropsten missing, fell back to mainnet v5). In the current code, if the user doesn't supply any v5 bootnodes, we'll leave them empty, making the light client non-functional.

**Pull request merged**